command/debug/debug.go

```diff
@@ -0,0 +1,613 @@
+		if err != nil {
+			return fmt.Errorf("failed to create compressed archive header: %s", err)
+		}
+
+		header.Name = filepath.Join(filepath.Base(c.output), strings.TrimPrefix(file, c.output))
+
+		if err := t.WriteHeader(header); err != nil {
+			return fmt.Errorf("failed to write compressed archive header: %s", err)
+		}
+
+		// Only copy files
+		if !fi.Mode().IsRegular() {
+			return nil
+		}
+
+		f, err := os.Open(file)
+		if err != nil {
+			return fmt.Errorf("failed to open target files for archive: %s", err)
+		}
+
+		if _, err := io.Copy(t, f); err != nil {
+			return fmt.Errorf("failed to copy files for archive: %s", err)
+		}
+
+		f.Close()
+
+		return nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to walk output path for archive: %s", err)
+	}
+
+	// Remove directory that has been archived
+	err = os.RemoveAll(c.output)
+	if err != nil {
+		return fmt.Errorf("failed to remove archived directory: %s", err)
+	}
+
+	return nil
+}
+
+// defaultTargets specifies the list of all targets that
+// will be captured by default
+func (c *cmd) defaultTargets() []string {
+	return append(c.dynamicTargets(), c.staticTargets()...)
+}
+
+// dynamicTargets returns all the supported targets
+// that are retrieved at the interval specified
+func (c *cmd) dynamicTargets() []string {
+	return []string{"metrics", "logs", "pprof"}
+}
+
+// staticTargets returns all the supported targets
+// that are retrieved at the start of the command execution
+func (c *cmd) staticTargets() []string {
+	return []string{"host", "agent", "cluster"}
+}
+
+func (c *cmd) Synopsis() string {
+	return synopsis
+}
+
+func (c *cmd) Help() string {
+	return c.help
+}
+
+const synopsis = "Monitors a Consul agent for the specified period of time, recording information about the agent, cluster, and environment to an archive written to the relative directory."
+const help = `
+Usage: consul debug [options]
+
+  Monitors a Consul agent for the specified period of time, recording
+  information about the agent, cluster, and environment to an archive
+  written to the relative directory.
+
+  If ACLs are enabled, an agent token must be supplied in order to perform
+  this operation.
+
+  To create a debug archive in the relative directory for the default
+  duration and interval, capturing all information available:
+
+      $ consul debug
+
+  Flags can be used to customize the duration and interval of the
+  operation. Note that the duration must be longer than the interval.
+
+      $ consul debug -interval=20s -duration=1m
+
+  By default, the archive containing the debugging information is
+  saved to the relative directory as a .tar.gz file. The
+  output path can be specified, as well as an option to disable
+  archiving, leaving the directory intact.
+
+      $ consul debug -output=/foo/bar/my-debugging -archive=false
+
+  Note: Information collected by this command has the potential
+  to be highly sensitive. We strongly recommend review of the
```

Comment:
Should we qualify that a bit more? If I read this I'd worry it means we are dumping ACL tokens and root keys and panic a bit about even trying it to see. I think it's great we are super conscious of the sensitivity of telemetry and network info but maybe we can give a better hint here to avoid unnecessary alarm!